tools/qc/fastq_pair/fastq_pair.xml

```diff
@@ -0,0 +1,93 @@
+<tool id="fastq_pair" name="Paired-end fastq pairer" profile="18.01" version="1.0+galaxy0">
+    <description>Rewrite paired-end fastq files to make sure that all reads have a mate and to separate out singletons.</description>
+    <requirements>
+        <requirement type="package" version="1.0">fastq-pair</requirement>
+    </requirements>
+    <command detect_errors="exit_code"><![CDATA[
+
+    ## Run fastq_pair
+    fastq_pair
+
+    #if $table_size:
+        -t '$table_size'
+    #end if
+
+    #if $print_elements:
+        -p '$print_elements'
+    #end if
+
+    #if $verbose:
+        '$verbose'
+    #end if
+
+    '$fastq1' '$fastq2' &&
+
+    #set fq1paired=$fastq1+".paired.fq"
+    #set fq2paired=$fastq2+".paired.fq"
+    #set fq1single=$fastq1+".single.fq"
+    #set fq2single=$fastq2+".single.fq"
+
+    mv '$fq1paired' 1.paired.fq &&
+    mv '$fq2paired' 2.paired.fq &&
+    mv '$fq1single' 1.single.fq &&
+    mv '$fq2single' 2.single.fq
+
+    ]]></command>
+    <inputs>
+        <param name="verbose" label="Verbose" optional="true" value="false" argument="-v" type="boolean" truevalue="-v" falsevalue="" checked="true" help="Verbose output. This is mainly for debugging." />
+        <param name="fastq1" label="FASTQ file 1" type="data" format="fastqsanger" optional="false" help="FASTQ file input, file 1 of pair" />
+        <param name="fastq2" label="FASTQ file 2" type="data" format="fastqsanger" optional="false" help="FASTQ file input, file 2 of pair" />
+        <param name="table_size" label="Table size" argument="-t" optional="true" value="100003" type="integer" help="Table size (default 100003)" />
+        <param name="print_elements" label="Number elements to print" argument="-p" type="integer" optional="true" help="Print this number of elements in each bucket in the table." />
+    </inputs>
+    <outputs>
+        <collection name="paired" type="paired" format_source="fastq1" label="${tool.name} on ${on_string}: Paired reads">
+            <data name="forward" from_work_dir="1.paired.fq" />
+            <data name="reverse" from_work_dir="2.paired.fq" />
+        </collection>
+        <collection name="singletons" type="paired" format_source="fastq1" label="${tool.name} on ${on_string}: Singletons">
+            <data name="forward" from_work_dir="1.single.fq" />
+            <data name="reverse" from_work_dir="2.single.fq" />
+        </collection>
```

Comment:
I don't think building a paired collection makes sense here if you already know that there will be zero overlap between the reads in those two datasets. I would go for either separate datasets or a list collection.

Reply:
I see, thanks again!